### ISSUE TITLE
[Core] long_running_node_failures failed in the master

### ISSUE THREAD

**Issue by rkooo567** [ray-project/ray#30588]
### What happened + What you expected to happen

<img width="1466" alt="Screen Shot 2022-11-22 at 8 21 00 AM" src="https://user-images.githubusercontent.com/18510752/203366725-07bec8c6-33b4-4e3f-a2ae-1a6a5d6ded92.png">

Seems to be flaky recently 

If I look at the recent failure, the script is killed unexpectedly 

### Versions / Dependencies

n/a

### Reproduction script

master

**Comment by scv119** [comment]
closes it for now. no useful logs can be downloaded, probably an infra issue.

**Comment by stephanie-wang** [comment]
Reopening as the test is still failing.

**Comment by xwjiang2010** [comment]
I am gonna drop release blocker here, as the most recent run on release branch is passing: https://buildkite.com/ray-project/release-tests-branch/builds/1222#0184bf27-89de-4a94-8e7f-18e0a28e00d5

**Comment by stephanie-wang** [comment]
Ah thanks, yes I should've dropped it earlier! We believe it's a testing issue anyway, not a bug in OSS.

**Comment by stephanie-wang** [comment]
Tried running this on master and it exits with 137, which is an OOM.

```bash
[INFO 2022-12-01 11:40:26,682] log.py: 36  Did not find any results.
[ERROR 2022-12-01 11:40:26,682] run_release_test.py: 157  Command returned non-success status: 137
Traceback (most recent call last):
  File "/home/swang/ray/release/ray_release/glue.py", line 275, in run_release_test
    command, env=command_env, timeout=command_timeout
  File "/home/swang/ray/release/ray_release/command_runner/sdk_runner.py", line 141, in run_command
    raise CommandError(f"Command returned non-success status: {status_code}")
ray_release.exception.CommandError: Command returned non-success status: 137

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "ray_release/scripts/run_release_test.py", line 153, in main
    no_terminate=no_terminate,
  File "/home/swang/ray/release/ray_release/glue.py", line 354, in run_release_test
    raise pipeline_exception
  File "/home/swang/ray/release/ray_release/glue.py", line 278, in run_release_test
    raise TestCommandError(e)
ray_release.exception.TestCommandError: Command returned non-success status: 137
[INFO 2022-12-01 11:40:26,682] run_release_test.py: 161  Release test pipeline for test long_running_node_failures completed. Returning with exit code = 40
```

The grafana for memory available looks like there could be a memory leak / instability (note the y-axis, the min value is 17.5GB, not 0):
![Screenshot from 2022-12-01 17-26-22](https://user-images.githubusercontent.com/2560516/205193887-41cb9490-a6d7-4007-9dbf-508dd24aafd6.png)

Definitely suspicious, but I'm not sure yet whether this is a regression and/or related to the failure. We don't have metrics saved anymore for a passing version of the test, so I'm rerunning on 2.1 wheels to see how the memory usage looked then.

**Comment by stephanie-wang** [comment]
It appears that the test failure is actually caused by a post-processing step taking too long. Should now be passing after #30850, and I opened another issue to look into why the post-processing step is taking longer than expected at #30859.

For now, I'll leave the 2.1 test running just to sanity-check, but most likely the memory usage is OK since it's still well above 0 at the end of the run.
